erpnext/controllers/stock_controller.py

```diff
@@ -845,6 +845,153 @@ def add_gl_entry(
 		gl_entries.append(self.get_gl_dict(gl_entry, item=item))
 
 
+@frappe.whitelist()
+def show_accounting_ledger_preview(company, doctype, docname):
+	frappe.db.savepoint("show_accounting_ledger_preview")
+
+	filters = {"company": company, "include_dimensions": 1}
+	doc = frappe.get_doc(doctype, docname)
+
+	gl_columns, gl_data = get_accounting_ledger_preview(doc, filters)
```

Comment:
Wrap these in try...except? 

Otherwise some write can be committed by mistake.

IMO no savepoint is required. Do a full rollback instead.